src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyThingAttachedToCertificate.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.usecases;
+
+import com.aws.greengrass.clientdevices.auth.api.Result;
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.exception.AuthenticationException;
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.Certificate;
+import com.aws.greengrass.clientdevices.auth.iot.IotAuthClient;
+import com.aws.greengrass.clientdevices.auth.iot.Thing;
+import com.aws.greengrass.clientdevices.auth.iot.dto.VerifyThingAttachedToCertificateDTO;
+import com.aws.greengrass.clientdevices.auth.iot.infra.ThingRegistry;
+
+import javax.inject.Inject;
+
+
+public class VerifyThingAttachedToCertificate
+        implements UseCases.UseCase<Boolean, VerifyThingAttachedToCertificateDTO> {
+    private final IotAuthClient iotAuthClient;
+    private final NetworkState networkState;
+    private final ThingRegistry thingRegistry;
+
+
+    /**
+     * Verify a certificate with IoT Core.
+     *
+     * @param iotAuthClient       IoT auth client
+     * @param thingRegistry       Thing Registry
+     * @param networkState        Network state
+     */
+    @Inject
+    public VerifyThingAttachedToCertificate(IotAuthClient iotAuthClient, ThingRegistry thingRegistry,
+                                  NetworkState networkState) {
+        this.iotAuthClient = iotAuthClient;
+        this.thingRegistry = thingRegistry;
+        this.networkState = networkState;
+    }
+
+    private boolean verifyLocally(Thing thing, Certificate certificate) {
+        return thing.isCertificateAttached(certificate.getCertificateId());
+    }
+
+    private boolean verifyFromCloud(Thing thing, Certificate certificate) {
```

Comment:
I think we need some logging in here. But that can be done separately.

Reply:
This is a copy past from what we had but I agree. More logs would be nice